cfg/optimize.go

```diff
@@ -229,9 +231,11 @@ func getOptimizedValue(
 	currentValue any,
 	profileName string,
 	machineType string,
+	input OptimizationInput,
```

Comment:
This argument should be optional to be in sync with the function ApplyOptimizations. I am not sure how it'll work otherwise. ALso, it's a new parameter, so it's safer design-wise to keep it as the last.




```suggestion
	input ...OptimizationInput,
```

Reply:
Passed as pointer, and nil represents no input.